File: WindowsApp1/DataLayer/BaseService.cs
using System.Collections.Generic;
using System.IO;
using System.Net;
using System.Text;
using Nancy.Json;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace DataLayer
{
    public class BaseService
    {
        public string ApiUrl { get; set; } = "https://localhost:5001/api";

        public string Read()
        {
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(ApiUrl);
            string responceString;
            request.Method = "GET";
            request.ContentType = "application/json";
            try
            {
                using (var result = request.GetResponse())
                {
                    using (var reader = new StreamReader(result.GetResponseStream()))
                    {
                        responceString = reader.ReadToEnd();
                    }
                }

                return responceString;
            }
            catch (WebException e)
            {
                var code = ((int)((HttpWebResponse)e.Response).StatusCode);
                if (code == 404)
                    return "Not Found";
                return "Error";
            }
            
        }

        public string Delete()
        {
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(ApiUrl);
            string responceString;
            request.Method = "DELETE";
            request.ContentType = "application/json";

            try
            {
                using (var result = request.GetResponse())
                {
                    using (var reader = new StreamReader(result.GetResponseStream()))
                    {
                        responceString = reader.ReadToEnd();
                    }
                }
                return responceString;
            }
            catch (WebException e)
            {
                var code = ((int)((HttpWebResponse)e.Response).StatusCode);
                if (code == 404)
                    return "Not Found";
                return "Error";
            }
            
        }

        public string Update<T>(T obj)
        {
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(ApiUrl);
            string responceString;
            string inputJson = new JavaScriptSerializer().Serialize(obj);

            
            var bytes = Encoding.UTF8.GetBytes(inputJson);

            request.Method = "POST";
            request.ContentType = "application/json";
            request.Accept = "application/json";
            request.ContentLength = bytes.Length;
            request.Expect = "application/json";
            request.GetRequestStream().Write(bytes, 0, bytes.Length);

            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
            {
                StreamReader reader;
                reader = new StreamReader(response.GetResponseStream());
                responceString = reader.ReadToEnd();
                reader.Close();
                response.Close();
            }

            var busObj = JsonConvert.DeserializeObject<T>(responceString).ToString();
            return busObj;
        }

        public List<string> Run<T>(T obj)
        {
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(ApiUrl);
            string inputJson = new JavaScriptSerializer().Serialize(obj);
            var bytes = Encoding.UTF8.GetBytes(inputJson);

            request.Method = "POST";
            request.ContentType = "application/json";
            request.Accept = "application/json";
            request.ContentLength = bytes.Length;
            request.Expect = "application/json";
            request.GetRequestStream().Write(bytes, 0, bytes.Length);

            try
            {
                using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
                {
                    StreamReader reader;
                    var rawlistresp = new List<string>();
                    reader = new StreamReader(response.GetResponseStream());
                    string rawresp = reader.ReadToEnd();
                    var array = JArray.Parse(rawresp);
                    reader.Close();
                    response.Close();
                    return array.ToObject<List<string>>();
                }
            }
            catch (WebException e)
            {
                var code = ((int)((HttpWebResponse)e.Response).StatusCode);
                if (code == 404)
                    return null;
                return null;
            }
        }
    }
}

File: WindowsApp1/DataLayer/BaseService.vb
Imports System.IO
Imports System.Net
Imports System.Text
Imports Nancy.Json
Imports Newtonsoft.Json.Linq

Public Class BaseService
    Public Property ApiUrl As String = "https://localhost:44317/api"

    Public Function GetApi(apiUrl As String) As String
        Dim responceString = String.Empty
        Dim request = CType(WebRequest.Create(apiUrl), HttpWebRequest)
        request.Method = "GET"
        request.ContentType = "application/json"

        Using result = request.GetResponse()
            Using reader = New StreamReader(result.GetResponseStream())
                responceString = reader.ReadToEnd()
            End Using
        End Using

        Return responceString
    End Function

    Public Function Read() As String
        Dim request = CType(WebRequest.Create(ApiUrl), HttpWebRequest)

        Dim responceString As String
        request.Method = "GET"
        request.ContentType = "application/json"

        Using result = request.GetResponse()
            Using reader = New StreamReader(result.GetResponseStream())
                responceString = reader.ReadToEnd()
            End Using
        End Using

        Return responceString
    End Function

    Public Function Delete() As String
        Dim request = CType(WebRequest.Create(ApiUrl), HttpWebRequest)

        Dim responceString As String
        request.Method = "DELETE"
        request.ContentType = "application/json"

        Using result = request.GetResponse()
            Using reader = New StreamReader(result.GetResponseStream())
                responceString = reader.ReadToEnd()
            End Using
        End Using

        Return responceString
    End Function

    Public Function Update(Of T)(obj As T) As String
        Dim request = CType(WebRequest.Create(ApiUrl), HttpWebRequest)

        Dim responceString As String
        Dim inputJson = (New JavaScriptSerializer()).Serialize(obj)
        Dim bytes As Byte() = Encoding.UTF8.GetBytes(inputJson)

        request.Method = "POST"
        request.ContentType = "application/json"
        request.Accept = "application/json"
        request.ContentLength = bytes.Length
        request.Expect = "application/json"
        request.GetRequestStream().Write(bytes, 0, bytes.Length)

        Using response As HttpWebResponse = DirectCast(request.GetResponse(), HttpWebResponse)
            Dim reader As StreamReader
            reader = New StreamReader(response.GetResponseStream())
            responceString = reader.ReadToEnd()
            reader.Close()
            response.Close()
            Return responceString
        End Using

    End Function

    Public Function Run(Of T)(obj As T) As JArray
        Dim request = CType(WebRequest.Create(ApiUrl), HttpWebRequest)

        Dim inputJson = (New JavaScriptSerializer()).Serialize(obj)
        Dim bytes As Byte() = Encoding.UTF8.GetBytes(inputJson)

        request.Method = "POST"
        request.ContentType = "application/json"
        request.Accept = "application/json"
        request.ContentLength = bytes.Length
        request.Expect = "application/json"
        request.GetRequestStream().Write(bytes, 0, bytes.Length)

        Using response As HttpWebResponse = DirectCast(request.GetResponse(), HttpWebResponse)
            Dim reader As StreamReader
            Dim rawlistresp As List(Of String) = New List(Of String)
            reader = New StreamReader(response.GetResponseStream())
            Dim rawresp = reader.ReadToEnd()
            Dim array As JArray = JArray.Parse(rawresp)
            reader.Close()
            response.Close()
            Return array
        End Using

    End Function
End Class
'Public Static Class ApiCall
'    {  
'        Public Static String GetApi(String ApiUrl)  
'        {  

'            var responseString = "";  
'            var request = (HttpWebRequest)WebRequest.Create(ApiUrl);  
'            request.Method = "GET";  
'            request.ContentType = "application/json";  

'            Using (var response1 = request.GetResponse())  
'            {  
'                Using (var reader = New StreamReader(response1.GetResponseStream()))  
'                {  
'                    responseString = reader.ReadToEnd();  
'                }  
'            }  
'            Return responseString;  

'        }  



'        Public Static String PostApi(String ApiUrl, String postData = "")  
'        {  

'            var request = (HttpWebRequest)WebRequest.Create(ApiUrl);  
'            var data = Encoding.ASCII.GetBytes(postData);  
'            request.Method = "POST";  
'            request.ContentType = "application/x-www-form-urlencoded";  
'            request.ContentLength = data.Length;  
'            Using (var stream = request.GetRequestStream())  
'            {  
'                stream.Write(data, 0, data.Length);  
'            }  
'            var response = (HttpWebResponse)request.GetResponse();  
'            var responseString = New StreamReader(response.GetResponseStream()).ReadToEnd();  
'            Return responseString;  
'        }  






'    }
